smarts/core/road_map.py

```diff
@@ -143,8 +143,8 @@ def features(self) -> List[RoadMap.Feature]:
         def features_near(self, pose: Pose, radius: float) -> List[RoadMap.Feature]:
             raise NotImplementedError()
 
-        def shape(self, buffer_width: float = 0.0) -> Polygon:
-            """Returns a convex polygon, buffered by width (which must be non-negative), around this surface."""
+        def shape(self, width: float, buffer_width: float = 0.0) -> Polygon:
+            """Returns a convex polygon, buffered by width + buffered_width (which must be non-negative), around this surface."""
```

Comment:
The lane width is varying for some lanes (like diverging and merging lanes) in `OpenDRIVE` map.
Initially when we want to increase the length of the width, we we doing it like this since lane width remains constant for SUMO.
``` python
  width = lane.width_at_offset(lane_offset)
  lane_shape = lane.shape(width + 0.3)
```
0.3 here is the increase in the width. But since lane width can be different at different offsets in a lane, we can never find the buffer (or increase or decrease in lane width) passed as a parameter. So we separated the buffer from width as a parameter to make it easier in `OpenDRIVE` to increase or decrease the width.

So the current code now is,
```python
 width = lane.width_at_offset(lane_offset)
 lane_shape = lane.shape(width, 0.3)
```

Reply:
The idea behind the `bufffer_width` parameter is that the resulting polygon would be that much bigger than the perimeter of the shape, no matter what the width of the shape is at any given point.  So for example, if a single lane starts out at 3 meters wide and at some point increases to 4 meters wide, and if you specify `buffer_width = 1`, then the returned polygon will start out at 5 meters wide and then increase to 6 meters wide at the appropriate point.  Crucially, the caller should not have to pass the width of the actual polygon at any point (which would be non-sensical anyway for non-rectangular polygons) but just the _buffer_ width around the perimeter.

This should be true no matter whether we're talking about rectangles or some other polygon shape.

So I still don't think 2 parameters are needed here (nor do I even think they make sense).

However, of course I may (still) be missing something.  So if you guys feel strongly this is required, then at a minimum, you need to fix the comment here such that anyone who is implementing this method in other extensions of `RoadMap` will know what each of these parameters is supposed to be.  (For example, I know of at least one proprietary extension of `RoadMap` already that will need to be updated for this parameter, but it's still unclear what would need to be changed there.)

Reply:
I still feel it should be there, otherwise we cannot compute the `buffer_width`  for one parameter. I will update the comments like you asked for.